core/ts/Poll.ts

```diff
@@ -707,26 +735,14 @@ export class Poll implements IPoll {
     while (msgs.length % messageBatchSize > 0) {
       msgs.push(msg.asCircuitInputs());
     }
-
     // we only take the messages we need for this batch
-    msgs = msgs.slice(index, index + messageBatchSize);
-
-    // insert zero value in the message tree as padding
-    while (this.messageTree.nextIndex < index + messageBatchSize) {
-      this.messageTree.insert(this.messageTree.zeroValue);
-    }
-
-    // generate the path to the subroot of the message tree for this batch
-    const messageSubrootPath = this.messageTree.genSubrootProof(index, index + messageBatchSize);
-
-    // verify it
-    assert(this.messageTree.verifyProof(messageSubrootPath), "The message subroot path is invalid");
+    msgs = msgs.slice((index - 1) * messageBatchSize, index * messageBatchSize);
```

Comment:
let's add a comment to explain this line please

Reply:
resolved